File: endless-loop.ts
// endless-loop.ts
const sleep = (ms: number) => new Promise<void>((r) => setTimeout(r, ms));

let running = true;

process.on("SIGINT", () => {
  console.log("\n🛑 SIGINT received. Shutting down...");
  running = false;
});

async function main() {
  let i = 0;

  while (running) {
    i++;

    // ✅ your repeating work goes here
    console.log("tick", i);

    // ✅ prevents CPU from pegging at 100%
    await sleep(250);
  }

  console.log("✅ loop ended cleanly");
}

main().catch((e) => {
  console.error("Fatal:", e);
  process.exit(1);
});

export {};

File: endless_loop.py
# endless_loop.py
import time
import signal

running = True

def handle_sigint(sig, frame):
    global running
    print("\n🛑 SIGINT received. Shutting down...")
    running = False

signal.signal(signal.SIGINT, handle_sigint)

i = 0
while running:
    i += 1
    print("tick", i)
    time.sleep(0.25)

print("✅ loop ended cleanly")
